main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
 	}
 
-	fmt.Println(localMasterHash.String())
-	fmt.Println(remoteMasterHash.String())
+	isLocalMasterInSyncWithRemoteMaster := localMasterHash.String() == remoteMasterHash.String()
+	if !isLocalMasterInSyncWithRemoteMaster {
+		fmt.Println("The local master branch is not in sync with the remote master branch. Must be in sync to conduct release process.")
+		return nil
+	}
+
+	fmt.Println("Finished prererelease checks.")
+	
+	fmt.Println("Guessing next release version...")
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
+
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
+
+	undoChanges := true
+	defer func() {
+		if undoChanges {
+			// git reset --hard origin/master
+			originMasterCommitHash, err := repository.ResolveRevision(plumbing.Revision(remoteMasterBranchName))
+			err = worktree.Reset(&git.ResetOptions{Mode: git.HardReset, Commit: *originMasterCommitHash })
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo local changes made for release '%s'. Please run 'git reset --hard origin/master' to undo manually.", nextReleaseVersion.String(), err)
+			}
+		}
+	}()
+
+	fmt.Println("Running prerelease scripts...")
+	preReleaseScriptsDirpath := path.Join(currentWorkingDirectory, relScriptsDirpath)
+	err = runPreReleaseScripts(preReleaseScriptsDirpath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while running prerelease scripts.")
+	}
+
+	fmt.Println("Updating the changelog...")
+	err = updateChangelog(changelogFilepath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while updating the changelog.")
+	}
+
+	fmt.Println("Committing changes locally...")
+	// Commit pre release changes
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
```

Comment:
Things you'll need to check: 

1) if you have untracked files (e.g. a new file), will it make it past the is-working-tree-clean check?
2) will this add untracked files?

Because I could see a world where the user has created a new file, it's untracked, we say "yep they don't have any staged or unstaged changes", and then we end up committing their untracked file here.

Reply:
According to the go-git API for [`IsClean()`](https://pkg.go.dev/github.com/go-git/go-git/v5#Status.IsClean) , it will only return true if all files are in `Unmodified` state. A new file is considered in an `Untracked` state and would not make it past the check. I double checked this in manual testing.